plugins/modules/foreman_compute_resource.py

```diff
@@ -71,78 +72,67 @@
 '''
 
 EXAMPLES = '''
-- name: vmware compute resource
+- name: Create livirt compute resource
   foreman_compute_resource:
     name: example_compute_resource
     locations:
       - Munich
     organizations:
       - ATIX
-    provider: vmware
+    provider: libvirt
     provider_params:
-      url: vsphere.example.com
-      user: admin
-      password: secret
-      datacenter: ax01
+      url: libvirt.example.com
+      display_type: vnc
     server_url: foreman.example.com
     username: admin
     password: secret
     state: present
 
-- name: ovirt compute resource
+- name: Update livirt compute resource
   foreman_compute_resource:
-    name: ovirt_compute_resource
+    name: example_compute_resource
+    description: updated compute resource
     locations:
-      - France/Toulouse
+      - Munich
     organizations:
-      - Example Org
-    provider: ovirt
+      - ATIX
+    provider: libvirt
     provider_params:
-      url: ovirt.example.com
-      user: ovirt-admin@example.com
-      password: ovirtsecret
-      datacenter: aa92fb54-0736-4066-8fa8-b8b9e3bd75ac
-      ovirt_quota: 24868ab9-c2a1-47c3-87e7-706f17d215ac
-      use_v4: true
+      url: libvirt.example.com
+      display_type: vnc
     server_url: foreman.example.com
     username: admin
     password: secret
     state: present
+
+- name: Delete livirt compute resource
+  foreman_compute_resource:
+    name: example_compute_resource
+    server_url: foreman.example.com
+    username: admin
+    password: secret
+    state: absent
 '''
 
 RETURN = ''' # '''
 
-try:
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_compute_resource,
-        find_organizations,
-        find_locations,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-
-    from nailgun.entities import (
-        AbstractComputeResource,
-        LibvirtComputeResource,
-        OVirtComputeResource,
-        VMWareComputeResource,
-    )
-
-    has_import_error = False
-except ImportError as e:
-    has_import_error = True
-    import_error_msg = str(e)
-
-from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'description': 'description',
-    'provider': 'provider',
-    'organizations': 'organization',
-    'locations': 'location',
-    'datacenter': 'datacenter',
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
+
+
+entity_spec = {
```

Comment:
with the merge of #376 this can be made simpler I guess?

Reply:
Modified, according to #376 .